### PR TITLE
Added count for unique models and components

### PR DESCRIPTION
**Pull request by abdullah1308**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@abdullah1308 will you like into the Lint Checks?

File: models/meshmodel/core/v1alpha1/component.go [hunk 1]
```diff
@@ -80,13 +80,24 @@ func CreateComponent(db *database.Handler, c ComponentDefinition) (uuid.UUID, er
...
 	type componentDefinitionWithModel struct {
 		ComponentDefinitionDB
 		ModelDB
 		CategoryDB
 	}
 
+	countUniqueComponents := func(components []componentDefinitionWithModel) int {
+		var set map[string]struct{}
+		for _, model := range components {
+			key := model.ComponentDefinitionDB.Kind + "@" + model.ModelDB.Name
```
Comment (leecalcote) [inline models/meshmodel/core/v1alpha1/component.go:93]: Is this how duplicity is checked, @harkiratsm? With a composite of these three items, @MUzairS15?

Comment (zakisk) [reply models/meshmodel/core/v1alpha1/component.go:93]: @abdullah1308 PR doesn't have link to issue where can I see issue?

**Comment by zakisk** [comment]
@abdullah1308 PR title doesn't have a link to the issue where can I see the issue?

**Comment by zakisk** [comment]
@abdullah1308 and can you please clarify to me, how the uniqueness of a component is defined. for example, if there is a slice of components like this `[A, C, B, A, D, F, E, C, P]` in your solution after loop iteration it ends up on the elements `[A, C, B, D, F, E, P]` contained in `set` map and returns  `len(set)` so is this desired uniqueness because the uniqueness can be evaluated as `[B, D, F, E, P]` since `A, C` are twice they aren't considered unique.

**Comment by abdullah1308** [comment]
> @abdullah1308 and can you please clarify to me, how the uniqueness of a component is defined. for example, if there is a slice of components like this `[A, C, B, A, D, F, E, C, P]` in your solution after loop iteration it ends up on the elements `[A, C, B, D, F, E, P]` contained in `set` map and returns `len(set)` so is this desired uniqueness because the uniqueness can be evaluated as `[B, D, F, E, P]` since `A, C` are twice they aren't considered unique.

@zakisk Yes, that is how it works

File: models/meshmodel/core/v1alpha1/component.go (hunk 1 above)
Comment (abdullah1308) [reply models/meshmodel/core/v1alpha1/component.go:93]: @leecalcote [meshmap reference](https://github.com/layer5labs/meshery-extensions/blob/37bd8a7e82e27c133b1cb4cce238cadcac44cbdd/meshmap/src/utils/hooks/useMeshModelSearchModels.js#L43-L65)

Comment (abdullah1308) [reply models/meshmodel/core/v1alpha1/component.go:93]: lint fixed

**Review by leecalcote — APPROVED** [review]

**Review by harkiratsm — COMMENTED** [review]
Do we require changes in the meshery server? // @abdullah1308

**Pull request merged**